src/Oscoin/Crypto/Address.hs

```diff
@@ -0,0 +1,111 @@
+{- | An address format for the Oscoin blockchain loosely based on Parity's SS58
+-- and on BTC's Base58Check.
+--
+-- See docs/address_format.md for an extensive documentation.
+--
+--}
+
+{-# LANGUAGE DataKinds            #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Oscoin.Crypto.Address
+    ( Address(addressPrefix, addressPayload)
+    , AddressPrefix
+    , AddressPayload
+
+    -- * Creating an Address
+    , fromPublicKey
+
+    -- * Parsing binary blobs back into addresses
+    , parseAddress
+
+    -- * Pretty-printing an Address
+    , renderAddress
+    , fmtAddress
+    ) where
+
+import           Oscoin.Prelude
+
+import qualified Oscoin.Configuration as Config
+import           Oscoin.Crypto.Address.Internal
+import           Oscoin.Crypto.Address.Serialisation as Serialisation
+import           Oscoin.Crypto.Hash
+import           Oscoin.Crypto.PubKey
+
+import qualified Codec.CBOR.ByteArray as CBOR
+import qualified Codec.Serialise as CBOR
+import           Data.ByteArray (ByteArrayAccess, convert)
+import           Data.ByteString.BaseN
+                 (decodeBase32z, encodeBase32z, encodedText)
+import           Data.Tagged
+import qualified Formatting as F
+
+{------------------------------------------------------------------------------
+    Creating an Address
+------------------------------------------------------------------------------}
+
+data AddressCreationError =
+    UnsupportedConfiguration (Config.Network, Config.Environment)
+    deriving Show
+
+-- | NOTE(adn) It's conceivable that we might want to crate a type to carry
```

Comment:
Can we move the note somewhere else and have a haddock for this?

Reply:
Perhaps it shouldn't be in the code, but just serves for the PR?

Reply:
Oh yeah, totally -- what I was planning to do was not to use this module for documentation purposes (as it's actually useful to keep technical notes in there) but rather write proper documentation at `docs/address_format.md`. How does that sound?

Reply:
84c5a8f

Reply:
Yeah, makes sense :+1: